report/main.tex

```diff
@@ -52,18 +63,171 @@ \section{Introduction}
 We used the ART paper~\cite{paranjape2023art}.
 
 
-\begin{ack}
-  Thanks Mom.
-\end{ack}
+\section{Background}
+
+
+\section{Methodology}
+%- provide tools that are analogous to long term memory in humans
+
+Our goal was to create a modified ART framework that allows for an LLM to read from, write to, and organize a form of long-term memory, without actually retraining model weights.
+This would allow for the model to dynamically access previously stored information, as well as update its local repository with valuable data as it reads through prompts.
+Given the finite context window of any LLM, ensuring that only relevant data is retrieved would be a crucial requirement of this system.
+With these goals and constraints in mind, we chose a Python dictionary as the underlying data structure for persisting information.
+A dictionary naturally organizes data into key-value pairs, which would naturally allow for organization and efficient retrieval based on the LLM selecting only relevant keys.
+The LLM would load the information stored at keys it finds relevant to its current task, and organize the information it stores by choosing an appropriate key.
+
+In the context of ART, this translates into the creation of three tools to accomplish our envisioned task. Firstly, we define \texttt{[list-keys]}, a tool that takes no inputs and returns a string containing all the keys in the dictionary.
+Next we define \texttt{[write]}, this tool requires a key-value pair and writes the pair into the dictionary, updating the value stored under the key if it is already present.
+Finally, we define \texttt{[read]}, a tool that takes a key as input and returns the information stored at that key in the dictionary.
+Currently the tools save the dictionary to, and read it from, storage to ensure the long term stability of the data.
+
+We demonstrate our tools by performing a summarization task on journal articles taken from DATASET HERE.
+As is standard with ART prompts, demonstrations of the task are prepended to the prompt to provide the LLM an example of how to handle the task~\cite{paranjape2023art}.
+Each article is split into multiple chunks that are fed one at a time into the modified ART framework, with the network choosing how to organize and persist data for later perusal.
+At the end of each article, each entry in the dictionary is then fed back into the LLM and a complete summary is generated. Experiments, results, and their analysis are discussed below.
```

Comment:
I think I expected the methods section to be a little more generic than this. I think I prefer leaving the specifics of the dataset (words like "article" and "summarization") to the Experiments section, and having the Methodology section focus on the theoretical idea of using tools to expand the range of things the model can make use of. Thoughts?

Reply:
what do you both mean by theoretical uses of tools?
I'll add to the background section a couple of sentences about tools being a way to get around LLM limitations - LLMs are famously bad at math and a tool could help with that...
@kylrth , is there something else Alex should add to Methodology section related to "theoretical idea of using tools " ?